blockchain/blockchain.go

```diff
@@ -1099,58 +1100,44 @@ type BlockSignFunc func(blockHash, stateDiffCommitment *felt.Felt) ([]*felt.Felt
 
 // Finalise will calculate the state commitment and block hash for the given pending block and append it to the
 // blockchain
-func (b *Blockchain) Finalise(pending *Pending, sign BlockSignFunc) error {
+func (b *Blockchain) Finalise(txs []core.Transaction, receipts []*core.TransactionReceipt, diff *core.StateDiff,
```

Comment:
what's the benefit of this change here?

Reply:
The goal was to make the function signature communicate what the builder needs to construct. Passing in `pending` feels a little more risky, since many of the fields are overwritten or ignored until the block hash calculation.

But I'll drop that commit for now in an effort to simplify the PR and save refactors for later.